### PR TITLE
Revamp of Clinical Reminders, version 3

### PR DESCRIPTION
**Pull request by bradymiller**
Ongoing project (based on bootstrap-updates code). Continued from following PRs:
1. https://github.com/openemr/openemr/pull/2872
2. tywrenn#11
3. bradymiller#12
4. https://github.com/tywrenn/openemr/pull/13
5. Next step is this PR!

This is a work in progress.

Up For Grabs demo for this PR is here:
https://www.open-emr.org/wiki/index.php/Development_Demo#Theta_-_Up_For_Grabs_Demo

### CODE REVIEW

File: interface/super/rules/base/template/basic.php [hunk 1]
```diff
@@ -6,24 +6,105 @@
...
+    }
+</script>
+
+
+<i class="fa fa-caret-<?php
+if ($setting_bootstrap_submenu == 'hide') {
+    echo 'down';
+} else {
+    echo 'up';
+} ?> menu_arrow" style="position:fixed;left:5px;top:5px;z-index:1099;" id="patient_caret" onclick='toggle_menu();' aria-hidden="true"></i>
```
Comment (tywrenn) [inline interface/super/rules/base/template/basic.php]: Use class `position-fixed`

File: interface/super/rules/controllers/alerts/view/list_actmgr.php [hunk 2]
```diff
@@ -12,89 +12,199 @@
 
 use OpenEMR\Gacl\GaclApi;
 
+$setting_bootstrap_submenu = prevSetting('', 'setting_bootstrap_submenu', 'setting_bootstrap_submenu', ' ');
 ?>
+<div class="title" style="display:none">
```
Comment (tywrenn) [inline interface/super/rules/controllers/alerts/view/list_actmgr.php]: Use `d-none` class

File: interface/super/rules/controllers/alerts/view/list_actmgr.php [hunk 3]
```diff
@@ -12,89 +12,199 @@
...
+                    <td>
+                        <a href="javascript:document.cdralertmgr.submit();" class="btn btn-sm btn-primary" onclick="top.restoreSession()"><span><?php echo xlt('Save'); ?></span></a>
+                        <a href="javascript:document.cdralertmgr.reset();" class="btn btn-sm btn-primary" onclick="top.restoreSession()"><span><?php echo xlt('Reset'); ?></span></a>
+                    </td>
+                </tr>
+            </table>
 
-&nbsp;
+            &nbsp;
+            <div style="display:block;" class="text-center">
```
Comment (tywrenn) [inline interface/super/rules/controllers/alerts/view/list_actmgr.php]: Use `d-block` class

File: interface/super/rules/controllers/alerts/view/list_actmgr.php [hunk 4]
```diff
@@ -12,89 +12,199 @@
...
+        <div class="col-12">
+            <div class="header">
+                <div class="title">
+                    <?php echo xlt('Clinical Reminders & Alerts Manager'); ?>
+                </div>
+                <div id="show_report" class="red">&nbsp;</div>
+            </div>
+        </div>
+        <div class="col-12">
+            <table class="">
```
Comment (tywrenn) [inline interface/super/rules/controllers/alerts/view/list_actmgr.php]: Remove unused class attribute?

File: interface/super/rules/controllers/alerts/view/list_actmgr.php [hunk 5]
```diff
@@ -12,89 +12,199 @@
...
+                                            $start_title = xl($aco_title);
+                                            $show_title =  mb_substr($start_title, 0, 25)."...";
+                                            echo "<option value='" . attr($section) . ":" . attr($aco) . "' " . $select . "> " . xlt($aco_section_title) . ": " . text($show_title)  . "</option>";
+                                        }
+                                    }
+
+                                        echo "</select>";
+                                    ?>
+                                </td>
+                                <td><input style="display:none" name="id[<?php echo attr($index); ?>]" value="<?php echo attr($rule->get_id()); ?>" /></td>
```
Comment (tywrenn) [inline interface/super/rules/controllers/alerts/view/list_actmgr.php]: Use `d-none` class

File: interface/super/rules/controllers/browse/view/list.php [hunk 6]
```diff
@@ -11,18 +11,26 @@
...
 ?>
-
 <script language="javascript" src="<?php js_src('list.js') ?>"></script>
 <script language="javascript" src="<?php js_src('jQuery.fn.sortElements.js') ?>"></script>
 
 <script type="text/javascript">
     var list = new list_rules();
     list.init();
 </script>
+<div class="title" style="display:none"><a href="<?php echo $GLOBALS['webroot']; ?>/interface/super/rules/index.php?action=browse!list"><?php
```
Comment (tywrenn) [inline interface/super/rules/controllers/browse/view/list.php]: Use `d-none` class

File: interface/super/rules/controllers/browse/view/plans_config.php [hunk 7]
```diff
@@ -320,84 +324,106 @@
...
-    <div id="cdr_mappings_form-div" class="cdr-form">
-        <div class="cdr-plans">
-            <?php echo xlt('Plan') . ':'; ?>
-            <select id="cdr-plans-select" name="cdr-plans-select" class="cdr-plans-select-class">
-                             <option id="select_plan" value="select_plan">- <?php echo xlt('SELECT PLAN'); ?> -</option>
-                <option id="divider" value="divider" disabled/>
-                <option id="add_new_plan" value="add_new_plan"><?php echo xlt('ADD NEW PLAN'); ?></option>
-            </select>
-            <input title="<?php echo xla('Delete Plan'); ?>" id="delete_plan" class="delete_button" type="image" style="display: none;"/>
+<div class="title" style="display:none"><a href="<?php echo $GLOBALS['webroot']; ?>/interface/super/rules/index.php?browse!plans_config"><?php
```
Comment (tywrenn) [inline interface/super/rules/controllers/browse/view/plans_config.php]: Use `d-none` class

File: interface/super/rules/controllers/browse/view/plans_config.php [hunk 8]
```diff
@@ -320,84 +324,106 @@
...
-        <div id="cdr_hide_show-div" style="display: none;">
-            <div id="plan_status_div" class="plan-status_div">
-                                <label id='plan-status-label'><?php echo xlt('Status') . ':'; ?></label>
-                <button id='cdr-status' disable><?php echo xlt('Activate'); ?></button>
+        <div class="cdr-mappings row text-center">
+            <div class="col-12 text-center">
+                <button id="delete_plan"
+                        class="btn btn-sm btn-danger icon_1X"
+                        title="<?php echo xla('Delete Plan'); ?>"
+                        style="display: none;"><i class="fa fa-trash-o"></i>
```
Comment (tywrenn) [inline interface/super/rules/controllers/browse/view/plans_config.php]: Use d-none class

File: interface/super/rules/controllers/browse/view/plans_config.php [hunk 9]
```diff
@@ -320,84 +324,106 @@
...
+                </button>
+                <div  class="plan-status_div">
+                </div>
+
             </div>
-            <br/>
+            <div id="cdr_mappings_form-div" class="cdr-form col-10 row text-left">
+
+                <div id="new_plan_container" class="col-6"></div>
+                <div id="cdr_hide_show-div" style="display: none;">
```
Comment (tywrenn) [inline interface/super/rules/controllers/browse/view/plans_config.php]: Use `d-none` class

File: interface/super/rules/controllers/detail/view/view.php [hunk 10]
```diff
@@ -1,239 +1,1367 @@
...
-        </p>
-
-        <?php if ($intervals->getTypes()) {?>
-        <p>
-            <div>
-                <span class="left_col colhead"><u><?php echo xlt('Type'); ?></u></span>
-                <span class="end_col colhead"><u><?php echo xlt('Detail'); ?></u></span>
+        <div class="col-12">
+            <div class="header">
+                <div class="title"><?php echo xlt('Clinical Reminder'); ?>: <i class="bolder"><?php echo xlt($rule->title); ?></i>  </div>
```
Comment (tywrenn) [inline interface/super/rules/controllers/detail/view/view.php]: If this is trying to make the font bolder, use `font-weight-bolder` class

File: interface/super/rules/controllers/detail/view/view.php [hunk 11]
```diff
@@ -1,239 +1,1367 @@
...
+                    <span class="title "><?php echo xlt('Summary'); ?> </span>
+                </div>
+                <div id="show_summary_1" class="col-12">
+                    <table class="table table-sm table-condensed text-left">
+                        <tr>
+                            <td class="text-right">
+                                <span class="underline"><?php echo xlt('Name'); ?>:</span>
+                            </td>
+
+                            <td colspan="3" class="table-100"><?php echo xlt($rule->title); ?></td>
```
Comment (tywrenn) [inline interface/super/rules/controllers/detail/view/view.php]: What is this `table-100` class?

File: interface/super/rules/controllers/detail/view/view.php [hunk 12]
```diff
@@ -1,239 +1,1367 @@
...
-                               class="action_link" onclick="top.restoreSession()">
-                                (<?php echo xlt('delete'); ?>)
-                            </a>
-                        </span>
-                        <span class="mid_col"><?php echo( text($criteria->getTitle()) ); ?></span>
-                        <span class="mid_col"><?php echo( text($criteria->getCharacteristics()) ); ?></span>
-                        <span class="end_col"><?php echo( text($criteria->getRequirements()) ); ?></span>
+            </div>
+        </div>
+        <div class="col-6" id="show_summary_edit" style="display: none;">
```
Comment (tywrenn) [inline interface/super/rules/controllers/detail/view/view.php]: Use `d-none` class

**Comment by bradymiller** [comment]
hi @tywrenn ,
Thanks for the code review. As an aside, are you interested in being a mentor for the google summer of code (OpenEMR was accepted for this today)? If so, send me an email to brady.g.miller@gmail.com . thanks! -brady

**Comment by juggernautsei** [comment]
Hi, I was looking at the up for grabs and there is not a menu link to get into the clinical rules. Where did it move to?

**Comment by stephenwaite** [comment]
hi @juggernautsei , it's admin->practice->alert manager

**Comment by bradymiller** [comment]
bump

**Comment by juggernautsei** [comment]
I have a requested feature for this module. The suggestion is to allow a range of CPT codes to be chosen instead of one by one. That is too time-consuming.

**Comment by bradymiller** [comment]
bump

**Comment by stephenwaite** [comment]
gonna def bump, not sure if @kchapple is interested in this overhaul as the `ecqm` commences, there's also #4050

**Comment by stephenwaite** [comment]
Bump, thanks @jug

**Pull request closed without merging**